enos/modules/upgrade_client/scripts/verify_allocs.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+set -euo pipefail
+
+error_exit() {
+    printf 'Error: %s' "${1}"
+    exit 1
+}
+
+MAX_WAIT_TIME=60  # Maximum wait time in seconds
+POLL_INTERVAL=2   # Interval between status checks
+
+elapsed_time=0
+last_error=
+client_id=
+
+checkClientReady() {
+    local client client_status
+    echo "Checking client health for $CLIENT_IP"
+
+    client=$(nomad node status -address "https://$CLIENT_IP:4646" -self -json) || {
+        last_error="Unable to get info for node at $CLIENT_IP"
+        return 1
+    }
+    client_status=$(echo "$client" | jq  -r '.Status')
+    if [ "$client_status" == "ready" ]; then
+        client_id=$(echo "$client" | jq '.ID' | tr -d '"')
+        last_error=
+        return 0
+    fi
+
+    last_error="Node at $CLIENT_IP is ${client_status}, not ready"
+    return 1
+}
+
+while true; do
+    checkClientReady && break
+    if [ "$elapsed_time" -ge "$MAX_WAIT_TIME" ]; then
+        error_exit "$last_error within $elapsed_time seconds."
+    fi
+
+    echo "$last_error within $elapsed_time seconds. Retrying in $POLL_INTERVAL seconds..."
+    sleep "$POLL_INTERVAL"
+    elapsed_time=$((elapsed_time + POLL_INTERVAL))
+done
+
+echo "Client $client_id at $CLIENT_IP is ready"
+
+# Quality: "nomad_alloc_reconect: A GET call to /v1/allocs will return the same IDs for running allocs before and after a client upgrade on each client"
+echo "Reading allocs for client at $CLIENT_IP"
+if ! current_allocs=$("nomad alloc status -json | jq -r --arg NODE_ID \"$(nomad node status -allocs -address https://$CLIENT_IP:4646 -self -json | jq -r '.ID')\" '[.[] | select(.ClientStatus == \"running\" and .NodeID == $NODE_ID) | .ID] | join(\" \")'"); then
+    error_exi "Failed to read allocs for node: $client_id"
+fi
+
+RUNNING_ALLOCS=$(nomad alloc status -json | jq -r --arg NODE_ID "$NODE_ID" '[.[] | select(.ClientStatus == "running" and .NodeID == $NODE_ID) | .ID] | join(" ")')
```

Comment:
We already have `$client_id`. Shouldn't we be using that instead of `$NODE_ID` and going back to the API to get it?

Reply:
you are right, I missed it